src/AccountManager.mo

```diff
@@ -0,0 +1,597 @@
+
+      // log possible changes in deposit/withdrawal minima
+      logMinimum(#deposit, minimumPrev.0);
+      logMinimum(#withdrawal, minimumPrev.1);
+
+      // log possible changes in deposit/withdrawal fee
+      logFee(#deposit, feePrev.0);
+      logFee(#withdrawal, feePrev.1);
+    };
+
+    /// Retrieves the sum of all current deposits.
+    public func depositedFunds() : Nat = depositRegistry.sum() + underwayFunds_;
+
+    /// Retrieves the sum of all current deposits.
+    public func underwayFunds() : Nat = underwayFunds_;
+
+    /// Retrieves the sum of all current deposits.
+    public func queuedFunds() : Nat = depositRegistry.sum();
+
+    /// Returns the size of the deposit registry.
+    public func depositsNumber() : Nat = depositRegistry.size();
+
+    /// Retrieves the sum of all successful consolidations.
+    public func totalConsolidated() : Nat = totalConsolidated_;
+
+    /// Retrieves the sum of all deductions from the main account.
+    public func totalWithdrawn() : Nat = totalWithdrawn_;
+
+    /// Retrieves the calculated balance of the main account.
+    public func consolidatedFunds() : Nat = totalConsolidated_ - totalWithdrawn_;
+
+    /// Retrieves the deposit of a principal.
+    public func getDeposit(p : Principal) : ?Nat = depositRegistry.getOpt(p);
+
+    func process_deposit(p : Principal, deposit : Nat, release : ?Nat -> Int) : Nat {
+      if (deposit <= fee(#deposit)) {
+        ignore release(null);
+        return 0;
+      };
+      let delta = release(?deposit);
+      if (delta < 0) freezeCallback("latestDeposit < prevDeposit on notify");
+      if (delta == 0) return 0;
+      let inc = Int.abs(delta);
+
+      if (deposit == inc) {
+        issue(p, deposit - fee(#deposit));
+      } else {
+        issue(p, inc);
+      };
+      inc;
+    };
+
+    /// Notifies of a deposit and schedules consolidation process.
+    /// Returns the newly detected deposit if successful.
+    public func notify(p : Principal) : async* ?Nat {
+      if (notificationsOnPause_) return null;
+      let ?release = depositRegistry.obtainLock(p) else return null;
+
+      let latestDeposit = try {
+        await* loadDeposit(p);
+      } catch (err) {
+        ignore release(null);
+        throw err;
+      };
+
+      if (latestDeposit < minimum(#deposit)) {
+        ignore release(null);
+        return ?0;
+      };
+
+      // This function calls release() to release the lock
+      let inc = process_deposit(p, latestDeposit, release);
+
+      if (inc > 0) {
+        log(p, #newDeposit(inc));
+
+        if (triggerOnNotifications) {
+          // schedule a canister self-call to initiate the consolidation
+          // we need try-catch so that we don't trap if scheduling fails synchronously
+          try { ignore async { await* trigger(1) } } catch (_) {};
+        };
+      };
+
+      return ?inc;
+    };
+
+    func processDepositTransfer(account : ICRC1.Account, amount : Nat) : async* {
```

Comment:
The function name is close to the `process_deposit` above. 

Maybe `process_allowance` is better to distinguish it?